sample_files/main.py

```diff
@@ -1,145 +1,80 @@
+from selenium import webdriver
+from webdriver_manager.chrome import ChromeDriverManager
+import logging
+import time
+import requests
+from requests.exceptions import RequestException
+from selenium.common.exceptions import WebDriverException
 from langchain_openai import ChatOpenAI
-import os
-import sys
-from langchain_core.prompts import ChatPromptTemplate
-from dotenv import load_dotenv
-from langchain_experimental.utilities import PythonREPL
-from langchain.tools import Tool
 from langchain_core.output_parsers import StrOutputParser
 from utils import sanitize_output, read_urls, html_scrapper
 from sys_prompts import system_prompt
-from logs.logger import logger
-import requests
-from requests.exceptions import RequestException
-import time 
-from selenium import webdriver
-from selenium.webdriver.common.by import By
-from selenium.common.exceptions import WebDriverException
-
-
-# Load environment variables from the .env file where our api key is stored
-load_dotenv()
-
-# fetching the api key from the .env file
-api_k = os.getenv('OPENAI_API_KEY')
-
-# chat model object (default model- gpt3.5 turbo)
-llm = ChatOpenAI(temperature=0.8,api_key=api_k)
-
+from logs.logger import logger  # Assuming this is a custom logger instance
 
-# Prompt for system message
-sys_prompt = system_prompt("Selenium 4.20.0 generator openai")
+# Initialize logging
+logging.basicConfig(level=logging.INFO)
 
+# Load environment variables
+from dotenv import load_dotenv
+import os
 
-# Fetching urls form config files
-page_list = ["admin_page", "data_entry"]
-urls = sys.argv[1:]
-
-
-# Web scrapping
-logger.info("Starting scraping... ")
-html_src_prompt = html_scrapper(urls)
-logger.info("Content scraped") 
-
-
-# Prompt
-prompt = ChatPromptTemplate.from_messages(
-    [
-        ("system", sys_prompt),
-        ("human", html_src_prompt),
-        ("human", "{urls}")
-    ]
-)
-
-# Setting up python_repl that runs valid python command
-python_repl = PythonREPL()
-repl_tool = Tool(
-    name = "python_repl",
-    description = '''A Python shell. Use this to execute python commands. 
-    Input should be a valid python command. Wait for code to execute completely ''',
-    func = python_repl.run,
-)
-
+load_dotenv()
 
-def save_and_process(output: str):
-    """Save output to a file and process it with repl_tool."""
-    # Save the sanitize_output to a file
-    if output is not None:
-        with open("output_scripts/selenium_script.py", "w+") as f:
-            f.write(output)
-    
-    # Process the output with the lambda function
-    return repl_tool(output) if output is not None else None
+# Fetch API key
+api_key = os.getenv('OPENAI_API_KEY')
 
- 
-# StrOutputParser() - converts response into parsable string
-chain = (
-        prompt |
-        llm | 
-        StrOutputParser() | 
-        sanitize_output |
-        save_and_process
-        )
+# Initialize OpenAI Chat model
+llm = ChatOpenAI(temperature=0.8, api_key=api_key)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The OpenAI Chat model is initialized without any error handling. If the API key is not found or is invalid, this will cause an unhandled exception. Consider adding error handling around this initialization.

```diff
- # Initialize OpenAI Chat model
- llm = ChatOpenAI(temperature=0.8, api_key=api_key)
+ try:
+     # Initialize OpenAI Chat model
+     llm = ChatOpenAI(temperature=0.8, api_key=api_key)
+ except Exception as e:
+     logger.error(f"Failed to initialize OpenAI Chat model: {e}")
+     sys.exit(1)
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->